.github/workflows/python.yml

```diff
@@ -0,0 +1,35 @@
+name: Python Code Quality
+
+on:
+  push:
+    branches: [main]
+  pull_request:
+    branches: [main]
+
+jobs:
+  check-python-code-quality:
+    runs-on: ubuntu-latest
+    timeout-minutes: 45
+
+    steps:
+      - name: Checkout Code
+        uses: actions/checkout@v4
+
+      - name: Set up Python
+        uses: actions/setup-python@v5
```

Comment:
You can use poetry with this action and it'll handle caching dependencies automatically for you. See [here](https://github.com/actions/setup-python/blob/main/docs/advanced-usage.md#caching-packages). Then you won't need `poetry install` below.

Reply:
example still has `poetry install`:

```
steps:
- uses: actions/checkout@v4
- name: Install poetry
  run: pipx install poetry
- uses: actions/setup-python@v4
  with:
    python-version: '3.9'
    cache: 'poetry'
- run: poetry install
- run: poetry run pytest
```